number_search.py

```diff
@@ -2,32 +2,33 @@
 #  in article.txt
 # use regex to match numbers and fractions
 import re
-frac_pattern = r'(-?\d+\\[A-Za-z]+{\d+}{\d+} |-?\\[A-Za-z]+{\d+}{\d+})'
-number_pattern = r"([^a-zA-Z{./\\}]\d+ |\d+\.\d+|[:]\d+|\d+[:])"
+frac_pattern = r'(-?\d+\\tinyfrac{\d+}{\d+}[^A-Za-z]|-?\\frac{\d+}{\d+}[^A-Za-z])'
+number_pattern = r'([^a-zA-Z{./\\}]\d+ |\d+\.\d+| \d+[^/:\\)-}])'
+numbers = []
+fractions = []
 
 
 def count_numbers(text):
     """
         counts whole number and decimals i.e -ve and +ve whole numbers decimals
         TODO update doc strings
     """
-    count = -1
+
     with open(text, "r") as f:
         article = f.read()
     pattern = re.compile(number_pattern)
     match = pattern.finditer(article)
     for x in match:
         print(x.group(0))
-        count += 1
-    return count
+        numbers.append(x.group(0))
```

Comment:
print this list you can see it is capturing newline.... which is not correct.. update you pattern